test/SparqlAntlrParserTest.cpp

```diff
@@ -782,3 +782,69 @@ TEST(SparqlParser, LimitOffsetClause) {
     EXPECT_EQ(limitOffset._remainingText, "Limit 20");
   }
 }
+
+TEST(SparqlParser, OrderCondition) {
+  // var
+  {
+    string input = "?test";
+    ParserAndVisitor p{input};
+    auto orderKey =
+        p.parser.orderCondition()->accept(&p.visitor).as<OrderKey>();
+    EXPECT_THAT(orderKey, IsVariableOrderKey("?test", false));
+  }
+  // brackettedExpression
```

Comment:
This should be a one-liner using `EXPECT_COMPLETE_PARSE` from the last PR (if this was the actual name, but maybe we have to extend this to additional arguments to the matcher to pass the `isDescending` information here.
Maybe first think about how to implement this yourself conceptually, and contact me if you have trouble with the strange details of C++ template magic, that's what I'm there for.

Reply:
There is only a `parse...` function for the whole orderClause but single orderConditions are tested here. If we want to use `expectCompleteParse` consistently we should maybe revisit the generalization of `parse...` before. Otherwise many `parse...` would be needed in SparqlParserHelpers.